### ISSUE TITLE
Why matrix H has 6 degrees of freedom?

### ISSUE THREAD

**Issue by andongchen** [MaybeShewill-CV/lanenet-lane-detection#216]
H=[a   b   c
      0   d  e
      o   f   1]
is f no use?

**Comment by MaybeShewill-CV** [comment]
@andongchen You may read chapter B. CURVE FITTING USING H-NET of the origin paper for details:)

**Comment by andongchen** [comment]
Sorry， I have read chapter B several times，but I still unknown where use the “f” in matrix H in curve fitting, Can you explain it in more detail, thank you very much .

In you paper you wrote:
Given a lane pixel pi = [xi; yi; 1]T belong to  P,
the transformed pixel p'i = [x'i; y'i; 1]T belong to P' is equal to Hpi=[axi+byi+c, dyi+e, fyi+1],
why p'i third element is one but not fyi+1? 

And in your loss funtion I still unkonw where use the pts_projects's third element?
`gt_pts = tf.transpose(gt_pts)
        pts_projects = tf.matmul(H, gt_pts)

        # 求解最小二乘二阶多项式拟合参数矩阵
        Y = tf.transpose(pts_projects[1, :])
        X = tf.transpose(pts_projects[0, :])
        Y_One = tf.add(tf.subtract(Y, Y), tf.constant(1.0, tf.float32))
        Y_stack = tf.stack([tf.pow(Y, 3), tf.pow(Y, 2), Y, Y_One], axis=1)
        w = tf.matmul(tf.matmul(tf.matrix_inverse(tf.matmul(tf.transpose(Y_stack), Y_stack)),
                                tf.transpose(Y_stack)),
                      tf.expand_dims(X, -1))
        # 利用二阶多项式参数求解拟合位置并反算到原始投影空间计算损失
        x_preds = tf.matmul(Y_stack, w)
        preds = tf.transpose(tf.stack([tf.squeeze(x_preds, -1), Y, Y_One], axis=1))
        x_transformation_back = tf.matmul(tf.matrix_inverse(H), preds)

        loss = tf.reduce_mean(tf.pow(gt_pts[0, :] - x_transformation_back[0, :], 2))`

**Comment by MaybeShewill-CV** [comment]
@andongchen That's not my paper:)
